### PR TITLE
[1.14] Jobs: change HTTP Create data request body to JSON

### PR DESCRIPTION
**Pull request by JoshVanL**
Previously, the HTTP Create data request body accepted only a fully qualified proto Any message- a object of type URL and Value. The API now accepts a JSON object string, i.e. number, string, boolean, object, etc. On trigger, the HTTP app will receive a JSON string of the Job data. In the cases where the data is not of type `google.protobuf.Value` or some other primitive encoded type, the data may contain binary data.

An internal `JobHTTPRequest` proto message has been added for the HTTP endpoint to allow of validation and ingestion of JSON string data in the request body.

The HTTP and gRPC channels now construct trigger job requests separately to allow for the different data serializations.

Closes https://github.com/dapr/dapr/issues/8077 https://github.com/dapr/dapr/issues/8076

Docs: https://github.com/dapr/docs/pull/4349

### CODE REVIEW

**Comment by yaron2** [comment]
Linter

**Comment by JoshVanL** [comment]
@yaron2 updated

**Review by cicoyle — DISMISSED** [review]
I tested this locally with my test-apps and it works like magic 🪄

File: tests/integration/suite/daprd/jobs/http/basic.go [hunk 1]
```diff
@@ -84,7 +84,8 @@ func (b *basic) Run(t *testing.T, ctx context.Context) {
 		for _, body := range []string{
 			`{"schedule": "@daily"}`,
 			`{"schedule": "@daily", "repeats": 3, "due_time": "10s", "ttl": "11s"}`,
-			`{"schedule": "@daily", "repeats": 3, "due_time": "10s", "ttl": "11s", "data": {"@type": "type.googleapis.com/google.protobuf.StringValue", "value": "Hello, World!"}}`,
+			`{"schedule": "@daily", "repeats": 3, "due_time": "10s", "ttl": "11s", "data": "{\"@type\": \"type.googleapis.com/google.protobuf.StringValue\", \"value\": \"Hello, World!\"}"}`,
```
Comment (artursouza) [inline tests/integration/suite/daprd/jobs/http/basic.go:87]: I thought we agreed not to have this format anymore. I am OK for now, but I vote for this to be removed for HTTP in 1.15.

Comment (JoshVanL) [reply tests/integration/suite/daprd/jobs/http/basic.go:87]: Yes. We don't ingest data payloads as proto objects and instead as JSON strings. as agreed.

It's still valid JSON though, and will be ingested as such. Please see the addition of quotations, and the next test case.

**Review by artursouza — APPROVED** [review]

**Pull request merged**